c/datasetjson.c

```diff
@@ -2517,6 +2525,391 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
 #endif /* __ZOWE_OS_ZOS */
 }
 
+static int configureTextunitsFromContentBody(JsonObject *object, int *configsCount, TextUnit **inputTextUnit) {
+  JsonProperty *currentProp = jsonObjectGetFirstProperty(object);
+  Json *value = NULL;
+  int parmDefn = DALDSORG_NULL;
+  int type = TEXT_UNIT_NULL;
+  int rc = 0;
+  while(currentProp != NULL){
+    value = jsonPropertyGetValue(currentProp);
+    char *valueString = jsonAsString(value);
+
+    if(valueString != NULL){
+      errno = 0;
+      int valueStrLen = strlen(valueString);
+      char *propString = jsonPropertyGetKey(currentProp);
+      if (!strcmp(propString, "dsorg")) {
+        parmDefn = (!strcmp(valueString, "PS")) ? DALDSORG_PS : DALDSORG_PO;
+        rc = setTextUnit(TEXT_UNIT_INT16, 0, NULL, parmDefn, DALDSORG, configsCount, inputTextUnit);
+      } else if(!strcmp(propString, "blksz")) {
+        long toi = strtol(valueString, NULL, 0);
+        if(errno != ERANGE){
+          if (toi <= 0x7FF8 && toi >= 0) { //<-- If DASD, if tape, it can be 80000000
+            type = TEXT_UNIT_INT16;
+          } else if (toi <= 0x80000000){
+            type = TEXT_UNIT_LONGINT;
+          }
+          if(type != TEXT_UNIT_NULL) {
+            rc = setTextUnit(type, 0, NULL, toi, DALBLKSZ, configsCount, inputTextUnit);
+          }
+        }
+      } else if(!strcmp(propString, "lrecl")) {
+        long toi = strtol(valueString, NULL, 0);
+        if (errno != ERANGE){
+          if (toi == 0x8000 || (toi <= 0x7FF8 && toi >= 0)) {
+            rc = setTextUnit(TEXT_UNIT_INT16, 0, NULL, toi, DALLRECL, configsCount, inputTextUnit);
+          }
+        }
+      } else if(!strcmp(propString, "volser")) {
+        if (valueStrLen <= VOLSER_SIZE){
+          rc = setTextUnit(TEXT_UNIT_STRING, VOLSER_SIZE, &(valueString)[0], 0, DALVLSER, configsCount, inputTextUnit);
+        }
+      } else if(!strcmp(propString, "recfm")) {
+        int setRECFM = 0;
+        if (indexOf(valueString, valueStrLen, 'A', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_A;
+        }
+        if (indexOf(valueString, valueStrLen, 'B', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_B;
+        }
+        if (indexOf(valueString, valueStrLen, 'V', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_V;
+        }
+        if (indexOf(valueString, valueStrLen, 'F', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_F;
+        }
+        if (indexOf(valueString, valueStrLen, 'U', 0) != -1){
+          setRECFM = setRECFM | DALRECFM_U;
+        }
+        rc = setTextUnit(TEXT_UNIT_CHAR, 0, NULL, setRECFM, DALRECFM, configsCount, inputTextUnit);
+      } else if(!strcmp(propString, "blkln")) {
+        long toi = strtol(valueString, NULL, 0);
+        if (errno != ERANGE) {
+          if (toi <= 0xFFFF || toi >= 0){
+            rc = setTextUnit(TEXT_UNIT_INT24, 0, NULL, toi, DALBLKLN, configsCount, inputTextUnit);
+          }
+        }
+      } else if(!strcmp(propString, "status")) {
```

Comment:
this should be called disp, not status, because disp is the term that everyone knows.

Reply:
resolved